src/uct/ugni/base/ugni_device.c

```diff
@@ -23,20 +25,13 @@
 typedef struct uct_ugni_job_info {
     uint8_t             ptag;                           /**< Protection tag */
     uint32_t            cookie;                         /**< Unique identifier generated by the PMI system */
-    int                 pmi_num_of_ranks;               /**< Number of ranks started by PMI */
-    int                 pmi_rank_id;                    /**< rank id assigned by PMI */
     int                 num_devices;                    /**< Number of devices */
     uct_ugni_device_t   devices[UCT_UGNI_MAX_DEVICES];  /**< Array of devices */
     int                 initialized;                    /**< Info status */
 } uct_ugni_job_info_t;
 
 static uct_ugni_job_info_t job_info = {
-    .ptag               = 0,
-    .cookie             = 0,
-    .pmi_num_of_ranks   = 0,
-    .pmi_rank_id        = 0,
     .num_devices        = -1,
-    .initialized        = 0,
 };
```

Comment:
Just as a note here. I removed any extraneous line setting a struct member to 0. C requires the compiler to initialize any uninitialized memory in the data sections to 0 (both public and private) so .ptag = 0 does nothing.